### PR TITLE
LPS-166901 - [Raylife AP] Create graph structure

### PR DESCRIPTION
**Pull request by pedroliveirav**
[Task: LPS-166901](https://issues.liferay.com/browse/LPS-166901)

### CODE REVIEW

**Comment by pedroliveirav** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		});
+	}, []);
+
+	policies?.map((policy) => {
+		if (!expirationPoliciesList[policy?.productName]) {
+			expirationPoliciesList[policy?.productName] = policy?.termPremium;
+		}
+		else {
+			expirationPoliciesList[policy?.productName] += policy?.termPremium;
+		}
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:91]:
```suggestion
		!expirationPoliciesList[policy?.productName] ?
			expirationPoliciesList[policy?.productName] = policy?.termPremium;
	:
			expirationPoliciesList[policy?.productName] += policy?.termPremium;
		
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+					.join('');
+			}
+
+			return name;
+		});
+
+		return productAbbrevation;
+	};
+
+	const formattedLabel = formatLabel(labelColumnsFormatted, 10);
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:115]: What does 10 mean? Try don't use magical numbers.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/layouts/3_policies/page-definition.json [hunk 3]
```diff
@@ -448,7 +447,7 @@
 																				"widgetConfig": {
 																					"properties": "route=policies-table"
 																				},
-																				"widgetName": "[$CLIENT_EXTENSION_ENTRY_ID:RAYLIFEAGENTPORTAL$]"
+																				"widgetName": "com_liferay_client_extension_web_internal_portlet_ClientExtensionEntryPortlet_20097_RAYLIFEAGENTPORTAL001"
```
Comment (joanagrego) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/layouts/3_policies/page-definition.json:450]:
```suggestion
																				"widgetName": "[$CLIENT_EXTENSION_ENTRY_ID:RAYLIFEAGENTPORTAL$]"
```

**Review by joanagrego — CHANGES_REQUESTED** [review]
Nice job! I added some suggestions :)

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css [hunk 4]
```diff
@@ -0,0 +1,19 @@
+.total-premium-due-container .total-premium-due-title::after {
+	background-color: var(--primary);
+	content: '';
+	display: block;
+	height: 1px;
+	margin-top: 38px;
+	position: absolute;
+	width: 19px;
+}
+.total-premium-due-container .lexicon-icon {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css:10]: Please, add a line break here.

```suggestion
}

.total-premium-due-container .lexicon-icon {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	font-size: 7px;
+}
+.bb-text {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css:13]: Please, add a line break here.
```suggestion
}

.bb-text {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/Products.js [hunk 6]
```diff
@@ -37,3 +37,19 @@ export function getChannelId(channelName) {
 		`${headlessAPI}/channels?filter=contains(name, '${channelName}')&fields=id`
 	);
 }
+
+export function getProductsNames() {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/Products.js:41]: I think it to be better to change the function name.

Does it make sense to you?

```suggestion
export function getProductNames() {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 7]
```diff
@@ -0,0 +1,183 @@
...
+				if (renewalDue < policyExpirationDays) {
+					policiesList.push(policy);
+				}
+			});
+
+			setPolicies(policiesList);
+
+			return policiesList;
+		});
+	}, []);
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:83]: There are 2 useEffect with the same dependency array. I think it to be better to create 1 useEffect and separate each request into functions. After this call each function into the useEffect.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 8]
```diff
@@ -0,0 +1,183 @@
...
+
+	policies?.map((policy) => {
+		if (!expirationPoliciesList[policy?.productName]) {
+			expirationPoliciesList[policy?.productName] = policy?.termPremium;
+		}
+		else {
+			expirationPoliciesList[policy?.productName] += policy?.termPremium;
+		}
+	});
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:92]: Please, change it. You are using a ```.map``` here. This should be used when you need to create a new array.

```suggestion
	policies?.forEach((policy) => {
		if (!expirationPoliciesList[policy?.productName]) {
			return expirationPoliciesList[policy?.productName] = policy?.termPremium;
		}
		
		return expirationPoliciesList[policy?.productName] += policy?.termPremium;
		
	});
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+		return productAbbrevation;
+	};
+
+	const formattedLabel = formatLabel(labelColumnsFormatted, 10);
+
+	const policiesValuesList = [...dataColumnsFormatted];
+	policiesValuesList.shift();
+
+	let sumOfPolicies = 0;
+
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js:121]: Please, remove this line.
```suggestion
```

**Review by silasbsi — CHANGES_REQUESTED** [review]
I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/components/bar-chart/index.js [hunk 10]
```diff
@@ -62,6 +65,18 @@ const BarChart = ({
 					columns: [labelColumns, dataColumns],
 					labels: {
 						colors: '#272833',
+						format: {
+							data: (x) => {
+								if (format) {
```
[the remark this answers is not in the document]
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/components/bar-chart/index.js:70]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+	font-size: 14px;
+}
```
[the remark this answers is not in the document]
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css:17]: Actually, it is necessary.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+	useEffect(() => {
+		getPoliciesChartExpiringPolicies(
+			currentDateString[0],
+			currentDateString[1],
+			'01',
+			nextMonthDate[0],
+			nextMonthDate[1],
+			nextMonthDate[2]
+		).then((results) => {
```
[the remark this answers is not in the document]
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 2 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css (hunk 4 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css (hunk 5 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.css]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 7 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 9 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

**Review by joanagrego — APPROVED** [review]
Approved! Thank you @pedroliveirav :white_check_mark:

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 8 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js (hunk 1 above)
Comment (pedroliveirav) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/policies-total-premium-due/index.js]: Done.

**Review by silasbsi — APPROVED** [review]

**Comment by silasbsi** [comment]
LGTM :heavy_check_mark: Thansk @pedroliveirav

**Pull request closed without merging**